homeassistant/components/lacrosse_view/sensor.py

```diff
@@ -0,0 +1,138 @@
+    """Mixin for required keys."""
+
+    value_fn: Callable[[Sensor, str], float]
+
+
+@dataclass
+class LaCrosseSensorEntityDescription(
+    SensorEntityDescription, LaCrosseSensorEntityDescriptionMixin
+):
+    """Description for LaCrosse View sensor."""
+
+
+PARALLEL_UPDATES = 0
+ICON_LIST = {
+    "Temperature": "mdi:thermometer",
+    "Humidity": "mdi:water-percent",
+    "HeatIndex": "mdi:thermometer",
+    "WindSpeed": "mdi:weather-windy",
+    "Rain": "mdi:water",
+}
+UNIT_LIST = {
+    "degrees_celsius": "°C",
+    "degrees_fahrenheit": "°F",
+    "relative_humidity": "%",
+    "kilometers_per_hour": "km/h",
+    "inches": "in",
+}
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up LaCrosse View from a config entry."""
+    coordinator: DataUpdateCoordinator[list[Sensor]] = hass.data[DOMAIN][
+        entry.entry_id
+    ]["coordinator"]
+    sensors: list[Sensor] = coordinator.data
+
+    sensor_list = []
+    for i, sensor in enumerate(sensors):
+        if not sensor.permissions.get("read"):
+            LOGGER.warning(
+                "No permission to read sensor %s, are you sure you're signed into the right account?",
+                sensor.name,
+            )
+            continue
+        for field in sensor.sensor_field_names:
+            sensor_list.append(
+                LaCrosseViewSensor(
+                    coordinator=coordinator,
+                    description=LaCrosseSensorEntityDescription(
+                        key=f"{i} {field}",
+                        device_class="temperature" if field == "Temperature" else None,
+                        name=f"{sensor.name} {sub(r'(?<!^)(?=[A-Z])', ' ', field)}"
+                        if field
+                        != sensor.name  # If the name is the same as the field, don't include it.
+                        else sub(r"(?<!^)(?=[A-Z])", " ", field),
+                        state_class=SensorStateClass.MEASUREMENT,
+                        value_fn=lambda sensor, field: float(
+                            sensor.data[field]["values"][-1]["s"]
+                        ),
+                        native_unit_of_measurement=UNIT_LIST.get(
+                            sensor.data[field]["unit"], None
+                        ),
+                    ),
+                )
+            )
+
+    async_add_entities(sensor_list)
+
+
+class LaCrosseViewSensor(
+    CoordinatorEntity[DataUpdateCoordinator[list[Sensor]]], SensorEntity
+):
+    """LaCrosse View sensor."""
+
+    sensor_index: int = 0
+    entity_description: LaCrosseSensorEntityDescription
+
+    def __init__(
+        self,
+        description: LaCrosseSensorEntityDescription,
+        coordinator: DataUpdateCoordinator[list[Sensor]],
+    ) -> None:
+        """Initialize."""
+        super().__init__(coordinator)
+        sensor = self.coordinator.data[int(description.key.split(" ")[0])]
```

Comment:
It would be a little more straight forward to just pass in the extra variables needed rather that coding information into the string and parsing it out, especially since the key is constructed right before `LaCrosseViewSensor` is created so the separate fields can be passed in.